src/gmail.js

```diff
@@ -961,6 +961,26 @@ var Gmail = function(localJQuery) {
             if (pathname && (pathname.endsWith("/i/s") || pathname.endsWith("/i/fd"))) {
                 api.tools.parse_request_payload(params, triggered);
             }
+            if (pathname && (pathname.endsWith("/i/s"))) {
+                if (params.body_params && params.body_params[2] && params.body_params[2][1] && params.body_params[2][1][0][2][2][7]) {
+                    // check params length  and get thread ids
+                    // console.log('params.body_params: ', params.body_params);
+                    if (params.body_params[2][1][0][2][2][7][2].length > 3) {
+                        const newThreadIds = params.body_params[2][1].map(obj => obj[2][1]);
+                        const newMessageIds = params.body_params[2][1].map(obj => obj[2][2][7][3]);
+                        // console.log(newMessageIds);
+                        // console.log('newThreadIds: ', newThreadIds);
+                        if (params.body_params[2][1][0][2][2][7][1][0] === "^k") {
+                            // console.log("delete_new");
+                            triggered["delete_new"] = [newThreadIds, newMessageIds];
+                        }
+                        if (params.body_params[2][1][0][2][2][7][1][0] === "^a") {
+                            // console.log("archive_new");
+                            triggered["archive_new"] = [newThreadIds, newMessageIds];
```

Comment:
The old event provides the following event-argiments: `[email_ids, params.url, params.body]`

We should probably strive to have a similar looking argument-list to these events too.

That leaves us the problem that in the "old" Gmail email-ID and thread-ID could be used interchangably, but now they no longer can (which has benefits and drawbacks).

How about we make this argument list: `[email_ids, params.url, params.body, thread_ids]`

That should at least make old style event-handlers compatible with the additional event metadata.

Or does anyone foresee any problems with that?